### PR TITLE
Dockerize SCEta + add custom metrics

### PR DESCRIPTION
**Pull request by steeevin88**
dockerized app.py, configuring Prometheus, adding metrics

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+COPY requirements.txt /server
+
+RUN pip install -r /server/requirements.txt
+
+COPY *.py .
+
+EXPOSE 8000
+
+ENTRYPOINT ["python", "app.py"]
```
Comment (evanugarte) [inline Dockerfile]: add new line to bottom of this file

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - 8000:8000
+  prometheus:
+    image: prom/prometheus:latest
+    restart: always
+    volumes:
+      - ./prometheus.yml:/etc/prometheus/prometheus.yml
+    command:
+      - '--config.file=/etc/prometheus/prometheus.yml'
+    ports:
+      - 9090:9090
```
Comment (evanugarte) [inline docker-compose.yml]: newline here too

**Review by evanugarte — COMMENTED** [review]
add newlines, looks good so far

File: metrics.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import enum
+
+import prometheus_client
+
+class Metrics(enum.Enum):
```
Comment (evanugarte) [inline metrics.py]: lets add a metric for
1. 511 api response codes
2. 511 api latency (to track how long the apis took to respond
3. cache last updated (we store a timestamp to this metric every time it updates)
4. http response code (increment with label each time we return 200/400 etc

File: docker-compose.yml (hunk 2 above)
Comment (steeevin88) [reply docker-compose.yml]: done

File: Dockerfile (hunk 1 above)
Comment (steeevin88) [reply Dockerfile]: done

File: app.py [hunk 4]
```diff
@@ -73,18 +77,25 @@ def update_cache():
...
+        # successful request to 511 API
+        MetricsHandler.api_response_codes.labels(response.status_code).inc() 
+
         content = response.content.decode('utf-8-sig')
         data = json.loads(content)
 
         all_incoming_buses = data.get('ServiceDelivery', {}).get('StopMonitoringDelivery', {}).get('MonitoredStopVisit')
         unique_buses = {}
         for bus in all_incoming_buses:
+            MetricsHandler.predictions_count.inc(1)
```
Comment (evanugarte) [inline app.py]:
```suggestion
            MetricsHandler.predictions_count.inc()
```

File: app.py [hunk 5]
```diff
@@ -101,7 +112,8 @@ def update_cache():
                 unique_buses[bus]
             )
             predictions.append(pred)
-            
+
+        MetricsHandler.stops_count.inc(1)
```
Comment (evanugarte) [inline app.py]:
```suggestion
        MetricsHandler.stops_count.inc()
```

Comment (evanugarte) [reply app.py]: from the docs

https://prometheus.github.io/client_python/instrumenting/counter/

File: metrics.py (hunk 3 above)
Comment (steeevin88) [reply metrics.py]: done

**Review by evanugarte — APPROVED** [review]

**Pull request merged**